satpy/readers/seviri_l1b_hrit.py

```diff
@@ -632,15 +668,48 @@ def get_area_def(self, dsid):
             ncols,
             nlines - upper_south_line,
             upper_area_extent)
-
         area = geometry.StackedAreaDefinition(lower_area, upper_area)
 
         self.area = area.squeeze()
-        return area
+        return self.area
 
     def get_dataset(self, key, info):
+        """Get the dataset."""
         res = super(HRITMSGFileHandler, self).get_dataset(key, info)
         res = self.calibrate(res, key.calibration)
+        if key.name == 'HRV' and self.fill_HRV:
```

Comment:
Maybe move the padding logic to a separate method, e.g. `pad_hrv_data`, and rename the existing `pad_hrv_data` to `_pad_hrv_data`?

Reply:
Ok